### PR TITLE
[SQL] Delete BeamSql class

### PR DESCRIPTION
**Pull request by akedin**
BeamSql doesn't add any value

------------------------

Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue, if applicable. This will automatically link the pull request to the issue.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

It will help us expedite review of your Pull Request if you tag someone (e.g. `@username`) to look at it.

### CODE REVIEW

**Comment by akedin** [comment]
R: @kennknowles @apilloud @amaliujia @XuMingmin @xumingming

**Comment by kennknowles** [comment]
I do like the change. I think most of our classes with `Beam` in them should just be simplified. Let's wait plenty of time for input, make sure we aren't going to have another big rename of this.

File: sdks/java/extensions/sql/src/main/java/org/apache/beam/sdk/extensions/sql/SqlTransform.java [hunk 1]
```diff
@@ -39,12 +40,44 @@
...
+ *
+ * //output the final result with TextIO
+ * outputTableB.apply(...).apply(TextIO.write().to("/my/output/path"));
+ *
+ * p.run().waitUntilFinish();
+ * }</pre>
  */
 @AutoValue
-public abstract class QueryTransform extends PTransform<PInput, PCollection<Row>> {
+@Experimental
```
Comment (amaliujia) [inline sdks/java/extensions/sql/src/main/java/org/apache/beam/sdk/extensions/sql/SqlTransform.java]: Just curious, why add @Experimental here?

Comment (akedin) [reply sdks/java/extensions/sql/src/main/java/org/apache/beam/sdk/extensions/sql/SqlTransform.java]: All Beam SQL is experimental at this moment, meaning that APIs can change at any time

**Pull request merged**